CCPA/Data Deletion Request Handling.md

```diff
@@ -0,0 +1,330 @@
+   </td>
+  </tr>
+  <tr>
+   <td>LG smartworld
+   </td>
+   <td>lg-app-store
+   </td>
+  </tr>
+</table>
+
+
+## Example Implementation
+
+Publisher site setup:
+
+```
+<html>
+  <head>
+    <script>
+      //API provided by the publisher or CMP:
+      function uspapi_addFrame()
+      {
+       if(!window.frames['__uspapiLocator'])
+       {
+        if(document.body)
+        {
+         var i = document.createElement('iframe');
+         i.style.cssText = 'display:none';
+         i.name = '__uspapiLocator';
+         document.body.appendChild(i);
+        }
+        else
+        {
+         window.setTimeout('uspapi_addFrame()', 10);
+        }
+       }
+      }
+      
+      function uspapi_stub()
+      {
+       var b = arguments;
+       __uspapi.a = __uspapi.a || [];
+       if(!b.length)
+       {
+        return __uspapi.a;
+       }
+       else
+       {
+        __uspapi.a.push([].slice.apply(b));
+       }
+      }
+      
+      function uspapi_msghandler(event)
+      {
+       var msgIsString = typeof event.data === 'string';
+       try
+       {
+        var json = msgIsString ? JSON.parse(event.data) : event.data;
+       }
+       catch(e)
+       {
+        var json = null;
+       }
+       if(typeof (json) === 'object' && json !== null && '__uspapiCall' in json)
+       {
+        var i = json.__uspapiCall;
+        window.__uspapi(i.command, i.version, function (retValue, success) 
+        {
+         var returnMsg = {
+         '__uspapiReturn': {
+          'returnValue': retValue,
+          'success'    : success,
+          'callId'     : i.callId
+          }
+         };
+         event.source.postMessage(msgIsString ? JSON.stringify(returnMsg) : returnMsg, '*');
+        }, i.parameter);
+       }
+      }
+      
+      uspapi_addFrame();
+      if(!('__uspapi' in window) || 
+         (typeof (window['__uspapi']) !== 'function' && 
+          typeof (window['__uspapi']) !== 'object' && 
+          (typeof (window['__uspapi']) === 'undefined' || 
+           window['__uspapi'] !== null)))
+      {
+       window['__uspapi'] = uspapi_stub;
+       window['__uspapi'].msgHandler = uspapi_msghandler;
+       if(window.addEventListener)
+       {
+        window.addEventListener('message', uspapi_msghandler, false);
+       }
+       else
+       {
+        window.attachEvent('onmessage', uspapi_msghandler);
+       }
+      }
+    </script>
+    <!-- include the vendor deletion scripts into the page: -->
+    <script src="https://vendorx.com/ccpa-delete-function.js"></script>
+  </head>
+  <body>
+    <button id="ccpa-delete">Delete my personal data</button>
+    <script>
+      document.getElementById('ccpa-delete').addEventListener('click', function ()
+      {
+       __uspapi('performDeletion', 1, null);
+      }, false);
+    </script>
+  </body>
+</html>
+```
+
+Vendor script (at `https://vendorx.com/ccpa-delete-function.js`)
+
+```
+// find the __uspapi frame
+if(!('__uspapi' in window)){
```

Comment:
@janwinkler did I do this correctly?

Reply:
Except for possibly my comment below it looks correct to me.